src/pages/signup/signup.jsx

```diff
@@ -1,9 +1,119 @@
+import axios from "axios";
+import React, { useReducer } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import "../../components/styles/main.css";
+import { signupFormReducer } from "../../reducer/authReducer";
+import { useAuth } from "../../context/authContext";
+
 const Signup = () => {
-    return (
-        <div>
-            <h1>Signup</h1>
-        </div>
-    )
-}
-
-export {Signup}
\ No newline at end of file
+  const navigate = useNavigate();
+  const { authDispatch } = useAuth();
+  const [{ firstName, lastName, email, password }, signupDispatch] = useReducer(
+    signupFormReducer,
+    { firstName: "", lastName: "", email: "", password: "" }
+  );
+  const submitHandler = async (e, firstName, lastName, email, password) => {
+    e.preventDefault();
+    try {
+      const response = await axios.post("api/auth/signup", {
+        firstName,
+        lastName,
+        email,
+        password,
+      });
+      console.log(response);
+      localStorage.setItem("token", response.data.encodedToken);
+      localStorage.setItem(
+        "userData",
+        JSON.stringify(response.data.createdUser)
+      );
+      authDispatch({ type: "ADD_TOKEN", payload: response.data.encodedToken });
+    } catch (err) {
+      console.log(err);
+    } finally {
+      navigate("/");
+    }
+  };
```

Comment:
you can put this in a try block so if signup is successful then only it will redirect to the home page.